pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1072,36 +1127,52 @@ func (n *NetworkPolicyController) syncAddressGroup(key string) error {
 	groupSelector := addressGroup.Selector
 	if groupSelector.Namespace != "" {
 		// Namespace presence indicates Pods must be selected from the same Namespace.
-		pods, _ = n.podLister.Pods(groupSelector.Namespace).List(groupSelector.PodSelector)
-	} else if groupSelector.NamespaceSelector != nil && groupSelector.PodSelector != nil {
-		// Pods must be selected from Namespaces matching nsSelector.
+		if groupSelector.PodSelector != nil {
+			pods, _ = n.podLister.Pods(groupSelector.Namespace).List(groupSelector.PodSelector)
+		} else if groupSelector.ExternalEntitySelector != nil {
+			externalEntities, _ = n.externalEntityLister.ExternalEntities(groupSelector.Namespace).List(groupSelector.ExternalEntitySelector)
+		}
+	} else if groupSelector.NamespaceSelector != nil && (groupSelector.PodSelector != nil || groupSelector.ExternalEntitySelector != nil) {
+		// Pods and ExternalEntities must be selected from Namespaces matching nsSelector.
 		namespaces, _ := n.namespaceLister.List(groupSelector.NamespaceSelector)
 		for _, ns := range namespaces {
-			nsPods, _ := n.podLister.Pods(ns.Name).List(groupSelector.PodSelector)
-			pods = append(pods, nsPods...)
+			if groupSelector.PodSelector != nil{
+				nsPods, _ := n.podLister.Pods(ns.Name).List(groupSelector.PodSelector)
+				pods = append(pods, nsPods...)
+			} else if groupSelector.ExternalEntitySelector != nil {
+				nsExtEntities, _ := n.externalEntityLister.ExternalEntities(ns.Name).List(groupSelector.ExternalEntitySelector)
+				externalEntities = append(externalEntities, nsExtEntities...)
+			}
 		}
 	} else if groupSelector.NamespaceSelector != nil {
 		// All the Pods from Namespaces matching the nsSelector must be selected.
 		namespaces, _ := n.namespaceLister.List(groupSelector.NamespaceSelector)
 		for _, ns := range namespaces {
 			nsPods, _ := n.podLister.Pods(ns.Name).List(labels.Everything())
 			pods = append(pods, nsPods...)
+			nsExtEntities, _ := n.externalEntityLister.ExternalEntities(ns.Name).List(labels.Everything())
+			externalEntities = append(externalEntities, nsExtEntities...)
 		}
 	}
 	podSet := networking.GroupMemberPodSet{}
+	entitySet := networking.GroupMemberSet{}
```

Comment:
nit, rename to memberSet? so that later on we can insert Pods to this set too?

Reply:
Done